server/app/reports/worker.py

```
@@ -0,0 +1,209 @@
+                            frequency=sub.frequency,
+                            period_key=period_key,
+                            recipients_override=recipients,
+                        )
+
+                    # Mark ledger success and update subscription last_sent_at
+                    ledger.status = "success"
+                    sub.last_sent_at = self._now_utc
+                    await db.commit()
+                    stats["sent_success"] += 1
+
+                except Exception as e:
```

Ruff (0.14.4)

Do not catch blind exception: `Exception`

(BLE001)